addons/godot-xr-tools/objects/snap_path.gd

```diff
@@ -0,0 +1,155 @@
+@tool
+extends XRToolsSnapZone
+class_name XRToolsSnapPath
+
+## An [XRToolsSnapZone] that allows [XRToolsPickable] to be placed along a child [Path3D] node.
+##     They can either be placed along any point in the curve, or at discrete intervals
+##     by setting "snap_interval" above 0.0. Note: Attached [XRToolsPickable]s will face the +Z axis
+
+
+## Real world distance between intervals in Meters.
+## Enabled when not 0
+@export  var snap_interval := 0.0:
+	set(v): snap_interval = absf(v)
+@onready var path : Path3D
+
+
+func _ready() -> void:
+	super._ready()
+	
+	for c in get_children():
+		if c is Path3D:
+			path = c
+			break
+
+
+func has_snap_interval() -> bool:
+	return !is_equal_approx(snap_interval, 0.0)
+
+
+func _get_configuration_warnings() -> PackedStringArray:
+	# Check for Path3D child
+	for c in get_children():
+		if c is Path3D:
+			path = c
+			return[]
+	return["This node has no path to place objects along. Consider adding a Path3D as a child to define its shape."]
+
+
+# Called when a target in our grab area is dropped
+func _on_target_dropped(target: Node3D) -> void:
+	# Skip if invalid
+	if !enabled or is_instance_valid(picked_up_object) or !is_instance_valid(target) or !path:
+		return
+	
+	# Make a zone that will destruct once its object has left
+	var zone   = _make_temp_zone()
+	var offset = _find_offset(path, target.global_position)
+	
+	# if snap guide
+	if _has_snap_guide(target):
+		# comply with guide
+		offset = _find_closest_offset_with_length(path.curve, offset, _get_snap_guide(target).length)
+		
+		# too large to place on path
+		if is_equal_approx(offset, -1.0):
+			return
+		
+	# if snap_interval has been set, use it
+	if has_snap_interval():
+		offset = snappedf(offset, snap_interval)
+	
+	# set position
+	zone.position = path.curve.sample_baked(offset)
+	
+	# Add zone as a child
+	path.add_child(zone)
+	zone.owner = path
+	
+	# Connect self-destruct with lambda
+	zone.has_dropped.connect(func(): zone.queue_free(), Object.ConnectFlags.CONNECT_ONE_SHOT)
+	
+	# Force pickup
+	if target.can_pick_up(self):
```

Comment:
Little nitpick but shouldn't you check this at the start and not bother with creating the temporary zone if we can't pickup the object?

Reply:
Yes, I missed that. I must've left it that way from when I first got it to work. Fixed in [472f0ae](https://github.com/GodotVR/godot-xr-tools/pull/685/commits/472f0aeea0441a519cef9fa28d4e5ad06096c22b)